haskell-decl-scan.el

```diff
@@ -356,20 +356,15 @@ there."
   (interactive)
   (haskell-ds-move-to-decl nil (haskell-ds-bird-p) nil))
 
-(defun haskell-ds-comment-p
-    (&optional
-     pt)
-  "Test if the cursor is on whitespace or a comment.
-
-`PT' defaults to `(point)'"
+(defun haskell-ds-comment-p ()
+  "Test if the cursor is on whitespace or a comment."
   ;; ensure result is `t' or `nil' instead of just truthy
   (if (or
        ;; is cursor on whitespace
-       (haskell-ds-whitespace-p (following-char))
+       (haskell-ds-whitespace-p)
        ;; http://emacs.stackexchange.com/questions/14269/how-to-detect-if-the-point-is-within-a-comment-area
        ;; is cursor at begging, inside, or end of comment
-       (let ((fontfaces (get-text-property (or pt
-                                               (point)) 'face)))
+       (let ((fontfaces (get-text-property (point) 'face)))
```

Comment:
I can't add comment on a line that follows, so I'm putting it here: comments in haskell have structure and we use more faces than just `font-lock-comment-face`: `haskell-pragma-face`, `haskell-liquid-haskell-annotation-face`, `haskell-literate-comment-face`. Sometimes we also add `(:weight bold)` and `(:slant italic)`.